runtime/bcutil/ClassFileOracle.cpp

```diff
@@ -2482,6 +2494,13 @@ ClassFileOracle::markConstantBasedOnCpType(U_16 cpIndex, bool assertNotDoubleOrL
 		}
 		_constantPoolMap->markConstantAsReferencedDoubleSlot(cpIndex);
 		break;
+	case CFR_CONSTANT_Dynamic:
+		if (assertNotDoubleOrLong) {
+			markConstantDynamicAsReferenced(cpIndex);
```

Comment:
Constant_Dynamic can create a long or a double.  Can you explain why the check is necessary here?

Reply:
for constantDynamic, this marking function should only be called from `ldc` and `ldc_w` which is not allowed to return a long or double. 
I ll double check with the static verify functionality if this check will ever be hit

Reply:
confirmed with @taliamccormick that static verify would catch the case when `ldc` or `ldc_w` has a double or long return type. I ll remove this check then